DiagnosticsExtension/Models/ConnectionStringValidator/FileShareStorageValidator.cs

```diff
@@ -0,0 +1,91 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="StorageValidator.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using DiagnosticsExtension.Controllers;
+using DiagnosticsExtension.Models.ConnectionStringValidator.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Queues;
+using Azure.Storage.Files;
+using Azure.Storage.Files.Shares;
+using Microsoft.WindowsAzure.Storage;
+
+namespace DiagnosticsExtension.Models.ConnectionStringValidator
+{
+    public class FileShareStorageValidator : IConnectionStringValidator
+    {
+        public string ProviderName => "Microsoft.WindowsAzure.Storage";
+
+        public ConnectionStringType Type => ConnectionStringType.FileShareStorageAccount;
+
+        ConnectionStringValidationResult.ManagedIdentityType identityType;
+        public async Task<ConnectionStringValidationResult> ValidateViaAppsettingAsync(string appsettingname, string entityName)
+        {
+            var response = new ConnectionStringValidationResult(Type);
+
+            try
+            {
+                var result = await TestConnectionStringViaAppSetting(appsettingname, entityName);
+                if (result.Succeeded)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Success;
+                }
+                else
+                {
+                    throw new Exception("Unexpected state reached: result.Succeeded == false is unexpected!");
+                }
+            }
+            catch (Exception e)
+            {
+                response = ConnectionStringResponseUtility.EvaluateResponseStatus(e, Type, identityType);
+            }
+
+            return response;
+        }
+
+        public async Task<TestConnectionData> TestConnectionStringViaAppSetting(string appSettingName, string entityName)
+        {
+            string value = "";
+            var envDict = Environment.GetEnvironmentVariables();
+            ShareServiceClient client = null;
+            try
+            {
+                if (envDict.Contains(appSettingName))
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new ShareServiceClient(value);
+                }
+                client.GetSharesAsync();
+            }
+            catch (ArgumentNullException e)
+            {
+                throw new EmptyConnectionStringException(e.Message, e);
+            }
+            catch (Exception e)
+            {
+                throw new MalformedConnectionStringException(e.Message, e);
```

Comment:
Why are we assuming this is the type of exception without examining the actual Exception that was thrown?